### PR TITLE
Override the service migration SDK version

### PR DESCRIPTION
**Pull request by tgjonestx**
Addresses Rally user stories US12315, US12316 and US12317

### CODE REVIEW

File: facade/service.go [hunk 1]
```diff
@@ -1517,15 +1518,26 @@ func executeMigrationScript(serviceID, tmpDir, scriptFilePath, inputFilePath str
...
+	if sdkVersion != "" {
+		tagName = sdkVersion
+	} else if tagOverride := os.Getenv("SERVICED_SERVICE_MIGRATION_TAG"); tagOverride != "" {
+		tagName = tagOverride
+	}
+
+	if tagName != "" {
+		glog.V(2).Infof("Facade:executeMigrationScript: using docker tag=%q", tagName)
+	}
+	dokckerImage := fmt.Sprintf("%s:%s", SERVICE_MIGRATION_IMAGE_NAME, tagName)
```
Comment (wplu) [inline facade/service.go]: dockerImage typo?

File: pkg/serviced.default [hunk 2]
```diff
@@ -126,7 +126,8 @@ SERVICED_FS_TYPE=btrfs
 # Default: 1    (enable)
 # SERVICED_MONITOR_DFS_MASTER_RESTART=1
 
+# Overrides the default for the service migration image.
+# SERVICED_SERVICE_MIGRATION_TAG=1.0.2
```
Comment (wplu) [inline pkg/serviced.default]: I'm not seeing this being used anywhere

Comment (tgjonestx) [inline pkg/serviced.default]: As I understand it, all of those settings are commented out by default, and the user can uncomment if they want to enable it.   The variable itself is referenced at line 1526 in facade/service.go
```
	} else if tagOverride := os.Getenv("SERVICED_SERVICE_MIGRATION_TAG"); tagOverride != "" {
		tagName = tagOverride
	}
```

File: facade/service.go [hunk 3]
```diff
@@ -1517,15 +1518,26 @@ func executeMigrationScript(serviceID, tmpDir, scriptFilePath, inputFilePath str
...
+
 	mountPath := fmt.Sprintf("%s:/migration", tmpDir)
 	cmd := exec.Command("docker",
 		"run", "--rm", "-t",
 		"--name", "service-migration",
 		"-v", mountPath,
-		SERVICE_MIGRATION_IMAGE_NAME,
+		dokckerImage,
```
Comment (wwwtyro) [inline facade/service.go]: Here too

File: script/nodes.go [hunk 4]
```diff
@@ -114,6 +114,32 @@ func buildNode(ctx *parseContext, cmd string, args []string) (node, error) {
...
+	//try to match
+	const SDK_VERSION_PATTERN = `^SDK=([a-zA-Z0-9.\-_]+)$`
+	sdkVerRegex := regexp.MustCompile(SDK_VERSION_PATTERN)
+	matches := sdkVerRegex.FindStringSubmatch(arg)
+	if len(matches) != 2 || matches[1] == "" {
+		return "", fmt.Errorf("arg %s did not match %s", arg, SDK_VERSION_PATTERN)
+	}
+	return matches[1], nil
+}
+
```
Comment (wwwtyro) [inline script/nodes.go]: Is there any reason we wouldn't want an exact match, here? i.e., just test "abc" == "abc" ?

**Comment by wwwtyro** [comment]
test this

File: pkg/serviced.default (hunk 2 above)
Comment (wplu) [inline pkg/serviced.default]: ahhh - must have been different commit

**Pull request merged**